v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/row/SerializableGenericRecordTest.java

```diff
@@ -0,0 +1,113 @@
+            .set(
+                "payload",
+                (new GenericRecordBuilder(schema.getAvroPayload()))
+                    .set("firstName", "abc")
+                    .set("lastName", "def")
+                    .build())
+            .build();
+    SerializableGenericRecord serializableGenericRecord = new SerializableGenericRecord(record);
+    assertThat(serializableGenericRecord.getRecord()).isEqualTo(record);
+  }
+
+  @Test
+  public void testSerde() throws IOException, ClassNotFoundException {
+    var schema = SchemaTestUtils.generateTestTableSchema("testTable");
+    GenericRecord genericRecord =
+        (new GenericRecordBuilder(schema.avroSchema()))
+            .set("read_timestamp", 1L)
+            .set(
+                "payload",
+                (new GenericRecordBuilder(schema.getAvroPayload()))
+                    .set("firstName", "abc")
+                    .set("lastName", "def")
+                    .build())
+            .build();
+    SerializableGenericRecord record = new SerializableGenericRecord(genericRecord);
+
+    /* Serialize */
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+    ObjectOutputStream outputStream = new ObjectOutputStream(bos);
+    outputStream.writeObject(record);
+    outputStream.flush();
+
+    /* Deserialize */
+    ByteArrayInputStream ios = new ByteArrayInputStream(bos.toByteArray());
+    ObjectInputStream inputStream = new ObjectInputStream(ios);
+    SerializableGenericRecord deSerializedRecord =
+        (SerializableGenericRecord) inputStream.readObject();
+
+    assertThat(record).isEqualTo(deSerializedRecord);
+  }
+
+  @Test
+  public void testEquality() {
+    var schema = SchemaTestUtils.generateTestTableSchema("testTable");
+    GenericRecord genericRecordLeft =
+        (new GenericRecordBuilder(schema.avroSchema()))
+            .set("read_timestamp", 1L)
+            .set(
+                "payload",
+                (new GenericRecordBuilder(schema.getAvroPayload()))
+                    .set("firstName", "abc")
+                    .set("lastName", "def")
+                    .build())
+            .build();
+    GenericRecord genericRecordRight =
+        (new GenericRecordBuilder(schema.avroSchema()))
+            .set("read_timestamp", genericRecordLeft.get("read_timestamp"))
+            .set("payload", genericRecordLeft.get("payload"))
+            .build();
+    SerializableGenericRecord recordLeft = new SerializableGenericRecord(genericRecordLeft);
+    SerializableGenericRecord recordRight = new SerializableGenericRecord(genericRecordRight);
+    assertThat(recordLeft).isEqualTo(recordRight);
+    assertThat(recordLeft.hashCode()).isEqualTo(recordRight.hashCode());
+    recordRight.getRecord().put(0, 4L);
+    assertThat(recordLeft).isEqualTo(recordLeft);
+    assertThat(recordLeft).isNotEqualTo(recordRight);
+    assertThat(recordLeft.hashCode()).isNotEqualTo(recordRight.hashCode());
+    assertThat(recordLeft).isNotEqualTo(123L);
```

Comment:
Any reason behind the 123 number?

Reply:
It's just to make UT coverage hit the check that left and right are of different type hence notEqual.